haproxy/errors/README.md

```diff
@@ -0,0 +1,4 @@
+# nginx_error_messages
+
+This directory contains error pages for haproxy. Do not update the error pages
+manually, instead look in `generate.sh` and the `template.json` file
```

Comment:
In general I prefer generated code to not be checked in to the repo, and instead generated in CI and published. This means someone doesn't accidentally manually update the output and saves on duplication. What do you think? Is there an advantage to checking in the build output?

Reply:
I can see arguments on both sides. The main reason I went with the more explicit pre-generated files is that I expect these will change at most 1-2 times per year and I didn't want to add complexity to the general build process. I imagine debugging an issue with the generated files without having been involved in this PR be more difficult if they're generated on build/start.

What if I add a test to ensure the current files are equal to how they will look when running generate.sh? I prefer that over build step as it doesn't complicate the build process and can be run independently from the build. I added a suggestion to what that could look like in this PR in the file `haproxy/tests/errors_generate.bats` feel free to review that one.